sklearn/utils/_array_api.py

```diff
@@ -466,6 +466,34 @@ def get_namespace_and_device(
         return xp, False, arrays_device
 
 
+def _convert_to_reference(*, reference, arrays):
```

Comment:
This is a nit/pet peeve: `sklearn.utils._array_api` is a private module, we don't need to make functions private here. It feels like repeating ourselves. I've made this argument before and you can tell I've not convinced everyone because there are many functions here that start with a `_` but I will keep trying :D

Reply:
That is a good point! I guess the `_` in the function name means that people easily know that this is a private function when seeing it used, but not knowing where it lives. Though I can't think of a good reason for needing to know this.

It is now `move_to` as @ogrisel suggested below.